odl/tomo/geometry/conebeam.py

```diff
@@ -993,248 +1003,210 @@ def __repr__(self):
-        # Get transformation and translation parts from `init_matrix`
-        init_matrix = np.asarray(init_matrix, dtype=float)
-        if init_matrix.shape not in ((3, 3), (3, 4)):
-            raise ValueError('`matrix` must have shape (3, 3) or (3, 4), '
-                             'got array with shape {}'
-                             ''.format(init_matrix.shape))
-        trafo_matrix = init_matrix[:, :3]
-        translation = init_matrix[:, 3:].squeeze()
-
-        # Transform the default vectors
-        default_axis = cls._default_config['axis']
-        default_src_to_det_init = cls._default_config['src_to_det_init']
-        default_det_axes_init = cls._default_config['det_axes_init']
-        vecs_to_transform = (default_src_to_det_init,) + default_det_axes_init
-        transformed_vecs = transform_system(
-            default_axis, None, vecs_to_transform, matrix=trafo_matrix)
-
-        # Use the standard constructor with these vectors
-        axis, src_to_det, det_axis_0, det_axis_1 = transformed_vecs
-        if translation.size == 0:
-            kwargs = {}
-        else:
-            kwargs = {'translation': translation}
+    References
+    ----------
+    .. [NW2001] Natterer, F and Wuebbeling, F.
+       *Mathematical Methods in Image Reconstruction*.
+       SIAM, 2001.
+       https://dx.doi.org/10.1137/1.9780898718324
+    """
+    # Find maximum distance from rotation axis
+    corners = space.domain.corners()[:, :2]
+    rho = np.max(np.linalg.norm(corners, axis=1))
+
+    # Find default values according to Nyquist criterion.
+
+    # We assume that the function is bandlimited by a wave along the x or y
+    # axis. The highest frequency we can measure is then a standing wave with
+    # period of twice the inter-node distance.
+    min_side = min(space.partition.cell_sides[:2])
+    omega = np.pi / min_side
+
+    # Compute minimum width of the detector to cover the object. The relation
+    # used here is (w/2)/(rs+rd) = rho/rs since both are equal to tan(alpha),
+    # where alpha is the half fan angle.
+    rs = float(src_radius)
+    rd = float(det_radius)
+    r = src_radius + det_radius
+    w = 2 * rho * (rs + rd) / rs
+
+    # Compute minimum number of pixels given the constraint on the
+    # sampling interval and the computed width
+    rb = np.hypot(r, w / 2)  # length of the boundary ray to the flat detector
+    num_px_horiz = 2 * int(np.ceil(w * omega * r / (2 * np.pi * rb))) + 1
+    delta_s = w / num_px_horiz
+
+    if space.ndim == 2:
+        det_min_pt = -w / 2
+        det_max_pt = w / 2
+        if det_shape is None:
+            det_shape = num_px_horiz
+    elif space.ndim == 3:
+        # Compute number of vertical pixels required to cover the object,
+        # using the same sampling interval vertically as horizontally.
+        # The reasoning is the same as for the computation of w.
+        corners = space.domain.corners()
+        rho_3d = np.max(np.linalg.norm(corners, axis=1))
```

Comment:
Shouldn't this only use the 3:rd axis? I.e. a very flat space shouldn't get a tall detector